### PR TITLE
Fixes #3: Update recommended approach for creating Solace ConnectionFactories enhancement

### LINKED ISSUE #3: Update recommended approach for creating Solace ConnectionFactories
In the basic samples it would be nice to update them to avoid the need to manually run CLI commands to create JNDI connection factories. Instead the samples should suggest using the `SolConnectionFactoryImpl` and creating it programmatically.  Here is some suggested sample code to get started.

```
            JMSProperties props;
            props = new JMSProperties((Hashtable<?,?>)null);
            props.initialize();
            SolConnectionFactoryImpl cf = new SolConnectionFactoryImpl(props);
            cf.setHost(properties.getHost());
            cf.setUsername(properties.getClientUsername());
            cf.setPassword(properties.getClientPassword());
            cf.setVPN(properties.getMsgVpn());
            cf.setDirectTransport(properties.isDirectTransport());  
            cf.setClientID(properties.getClientName());
```

### PR DESCRIPTION
**Pull request by bczoma**
Updated both samples and documentation.

### CODE REVIEW

File: docs/_config.yml [hunk 1]
```diff
@@ -15,6 +15,9 @@ docs-vmr-setup: http://docs.solace.com/Solace-VMR-Set-Up/Setting-Up-VMRs.htm
 docs-core-concepts: http://docs.solace.com/Features/Core-Concepts.htm
 docs-jms-home: http://docs.solace.com/Solace-JMS-API/JMS-home-open.htm
 docs-jms-connections: http://docs.solace.com/Solace-JMS-API/Establishing-Connections.htm
+docs-jms-obtaining_connection_factories: http://docs.solace.com/Solace-JMS-API/Obtaining-Connection-Fac.htm
+docs-jms-working_with_destinations: http://docs.solace.com/Solace-JMS-API/Working-with-Destination.htm
+docs-management_tools: http://docs.solace.com/Management-Tools.htm
 docs-gm-rr: http://docs.solace.com/Solace-JMS-API/Using-Topic-Requestors.htm
```
Comment (mdspielman) [inline docs/_config.yml]: For these links, please follow the convention of dash separated names. I.e. replace the underscores.

File: docs/_tutorials/persistence-with-queues.md [hunk 2]
```diff
@@ -76,93 +76,56 @@ This tutorial is available in [GitHub]({{ site.repository }}){:target="_blank"}
...
-    </tr>
-    <tr>
-        <td>Queue</td>
-        <td>/JNDI/Q/tutorial</td>
-    </tr>
-</table>
+A difference to the publish/subscribe tutorial is that here an additional physical endpoint resource – a durable queue, associated with the Queue Destination – is required on the message router, which will persist the messages until consumed.
 
-The script also creates a durable queue named “Q/tutorial” and enables this queue. This script assumes the connection factory and queue do not exist and creates them. If the JNDI connection factory or queue already exists you may need to remove the keyword “create” from the script below.
+Using the Solace Dynamic Durables property enables the physical endpoint to be dynamically created if it doesn’t exist yet, when the JMS Queue Destination is created programmatically. For other ways of creating Solace message router resources, refer to the [Solace Documentation - Management Tools]({{ site.docs-management_tools }}){:target="_top"}
```
Comment (mdspielman) [inline docs/_tutorials/persistence-with-queues.md]: Can you move this comment below to where you show how dynamic durables work.

File: docs/_tutorials/persistence-with-queues.md [hunk 3]
```diff
@@ -76,93 +76,56 @@ This tutorial is available in [GitHub]({{ site.repository }}){:target="_blank"}
...
 
+Here we create a queue from the JMS Session programmatically. For other ways of obtaining a queue, for example using JNDI, refer to the [Solace Documentation - Working with Destinations]({{ site.docs-jms-working_with_destinations }}){:target="_top"}.
+
+Because the Dynamic Durables property has been enabled when creating the connection, it will ensure that the physical queue endpoint resource will also be created dynamically if it doesn’t exist.
+
+```java
+Queue queue = session.createQueue("Q/tutorial");
+```
+
+Next, the JMS MessageProducer  is created from the JMS Session.
```
Comment (mdspielman) [inline docs/_tutorials/persistence-with-queues.md]: Extra space between "MessageProducer" and "is"

**Review by mdspielman — CHANGES_REQUESTED** [review]
Generally looks good. Can you address the small findings prior to merging.

File: docs/_tutorials/persistence-with-queues.md (hunk 2 above)
Comment (bczoma) [reply docs/_tutorials/persistence-with-queues.md]: The thinking was to introduce the notion of Dynamic Durables by describing the need to create a physical resource on the router and Dynamic Durables being the answer to that. So it is clear in the next sections (Connecting a session to the message router and Sending a message to a queue) what we are talking about. Do you think it is better to separate the two paragraphs below and move the second one under Sending a message to a queue ? We should have some kind of explanation why we want to enable it for the Connection Factory...

"[currently in section Solace message router JMS administered objects]
A difference to the publish/subscribe tutorial is that here an additional physical endpoint resource – a durable queue, associated with the Queue Destination – is required on the message router, which will persist the messages until consumed.

Using the Solace Dynamic Durables property enables the physical endpoint to be dynamically created if it doesn’t exist yet, when the JMS Queue Destination is created programmatically. For other ways of creating Solace message router resources, refer to the Solace Documentation - Management Tools"

Comment (mdspielman) [reply docs/_tutorials/persistence-with-queues.md]: I would suggest using the pattern of: 

1) This tutorial does X
2) Here's why doing X is sensible and here are alternatives.

That's my preference.

Comment (mdspielman) [reply docs/_tutorials/persistence-with-queues.md]: An alternative approach would be:

1) Here is an interesting feature X. It has the following properties.
2) This tutorial will make use of X and here is the code for how to do it.

That's also fine. IMO, the original text is half way between both approaches.

**Comment by bczoma** [comment]
Committed changes addressing Mark's change requests.

**Pull request merged**